video-player-in-flutter-using-chewie-and-getx/index.md

```diff
@@ -0,0 +1,287 @@
+The default flutter `video_player` limits the user when they want to perform an action to a video. Chewie is an Application Programming Interface that adds more functionality to the default flutter video player. Some of these functions include are like the addition of play controls to the video player.
+
+In this article, we're going to build a video player using Dart and Flutter.
+
+### Key Takeaways
+
+- What is `Chewie?`
+- Learn how to use `Chewie`.
+- Implementation of `Chewie` in a Flutter app.
+- Code Implementation of `Chewie`.
+
+We'll be making use of two libraries:
+
+* `Chewie`, the flutter plugin for decoding, playback, and camera. Build amazing videos with access to the frame by frame control and camera view.Chewie is a fully customizable video player which can play/stream almost any kind of format and provides an easy to use api.
+
+* `GetX`, a dart library for downloading files from the internet. It makes it very easy to download from a URL or a file into an input stream.Getx is a wrapper around the low level android & ios media APIs.Like requestPermission , getToken etc. You can use it to play videos too or capture photos.
+
+We'll take advantage of the first library, `Chewie` to play videos and audio files, and the second one, `GetX`, for downloading them from Youtube's  API.
+
+### Prerequisites
+
+- Visual Studio or any code editor installed.
+- Have an Understanding of Flutter Widgets
+- Familiar with dart programming languages.
+- Knowledge in Flutter.
+
+We're going to create a screen to show our video player. This screen will have buttons for controlling the playback and two text fields to write the URL of the resource we want to play and its title.Our videos will be fetched from the internet rather than playing the local videos in our phones.
+
+#### VideoScreen
+This is the homepage of our Video player application.This is the page that we will use to interact with the our videos.Let's start with the a basic page with a title.Our Title will be positioned at the top of our page and will be contained in the Appbar Widget.
+
+```dart 
+
+class VideoScreen extends StatefulWidget { 
+
+@override
+
+VideoScreenState createState() => new VideoScreenState();
+
+} 
+
+class VideoScreenState extends State { 
+
+@override 
+
+Widget build(BuildContext context) { 
+
+ return new Scaffold(//contains our components of the application.
+
+     appBar: new AppBar( 
+
+       title: new Text("Video Player"), 
+
+     ),
+
+   );
+
+ } 
+
+}
+
+```
+
+As you see, we have a simple screen with an `appbar `, which will have the title of our player, and a `Scaffold`. The `Scaffold` will contain the background image of our video player and two rows: one with the "Play" button on top and another showing the title and URL of the selected resource.Play button will be used to execute the streaming of our video.The Title and URL are just simply for identification purposes of our video just to be sure which video is playing at that time.
```

Comment:
Change to "As you see, we have a simple screen with an `appbar `, which will have the title of our player, and a `Scaffold`. The `Scaffold` will contain the background image of our video player and two rows: one with the "Play" button on top and another showing the title and URL of the selected resource. We will use the play button to execute the streaming of our video. The `Title` and `URL` are simply for identification purposes of our video to be sure which video is playing at that time."